syntax = "proto3";

package armonik.api.grpc.v1.results;

import "google/protobuf/timestamp.proto";
import "order_direction.proto";
import "result_status.proto";

option csharp_namespace = "ArmoniK.Api.gRPC.V1.Results";

/**
 * A raw result object.
 *
 * Used when a list or a single result is returned.
 */
message ResultRaw {
  string session_id = 1; /** The session ID. */
  string name = 2; /** The result name. */
  string owner_task_id = 3; /** The owner task ID. */
  result_status.ResultStatus status = 4; /** The result status. */
  google.protobuf.Timestamp created_at = 5; /** The result creation date. */
}

/**
 * Request to list results.
 *
 * Use pagination, filtering and sorting.
 */
message ListResultsRequest {
  int32 page = 1; /** The page number. Start at 0. */
  int32 page_size = 2; /** The page size. */

  string order_by = 3; /** The order by. */
  order_direction.OrderDirection order = 4; /** The order. */

  map<string, string> filters = 5; /** The filters. */
}

/**
 * Response to list results.
 *
 * Use pagination, filtering and sorting from the request.
 * Retunr a list of raw results.
 */
message ListResultsResponse {
  repeated ResultRaw results = 1; /** The list of raw results. */

  int32 page = 2; /** The page number. Start at 0. */
  int32 page_size = 3; /** The page size. */
  int32 total = 4; /** The total number of results. */
}

/*
*  Request for getting the id of the task that should create this result
*/
message GetOwnerTaskIdRequest {
  string session_id = 1;
  repeated string result_id = 2;
}

/*
* Response for getting the id of the task that should create this result
*/
message GetOwnerTaskIdResponse {
  message MapResultTask {
    string result_id = 1;
    string task_id = 2;
  }
  repeated MapResultTask result_task = 1;
  string session_id = 2;
}
